src/main/java/tree/BinarySearchTree.java

```diff
@@ -0,0 +1,225 @@
+        }
+        else {
+            if (root.getRight() == null){
+                root.setRight(newNode);
+                return root;
+            }
+            return addHelper(root.getRight(), newNode);
+        }
+    }
+
+    public boolean search (int data){
+        return searchHelper(this.root, data);
+    }
+
+    private boolean searchHelper(BSTNode root, int data){
+        if (root == null){
+            return false;
+        }
+        else if (root.getData() == data){
+            return true;
+        } else if (root.getData() >= data) {
+            return searchHelper(root.getLeft(), data);
+        } else {
+            return searchHelper(root.getRight(), data);
+        }
+    }
+    public int findMin(){
+        BSTNode currNode = this.root;
+        while (currNode.getLeft() != null){
+            currNode = currNode.getLeft();
+        }
+        return currNode.getData();
+    }
+    public int findMax(){
+        BSTNode currNode = this.root;
+        while(currNode.getRight() != null){
+            currNode = currNode.getRight();
+        }
+        return currNode.getData();
+    }
+    public int findHeight(){
+        return findHeightHelper(this.root);
+    }
+    private int findHeightHelper(BSTNode node){
+        if (node == null){
+            return 0;
+        }
+        int leftHeight = findHeightHelper(node.getLeft());
+            int rightHeight = findHeightHelper(node.getRight());
+        return leftHeight > rightHeight ? leftHeight + 1 : rightHeight + 1;
+    }
+
+    public void levelOrderTraversal(){
+        System.out.println("Level order traversal");
+        BSTNode currNode = this.root;
+        Queue queue = new LinkedList<BSTNode>();
+        queue.add(currNode);
+        while (!queue.isEmpty()){
+            currNode = (BSTNode) queue.poll();
+            System.out.print(currNode.getData() + " ");
+            if (currNode.getLeft() != null){
+                queue.add(currNode.getLeft());
+            }
+            if (currNode.getRight() != null){
+                queue.add(currNode.getRight());
+            }
+        }
+        System.out.println("\n");
+    }
+
+    public void preOrderTraversal(){
+        System.out.println("Pre order traversal");
+        preOrderTraversalHelper(this.root);
+        System.out.println("\n");
+    }
+    private void preOrderTraversalHelper(BSTNode root){
+        if (root == null){
+            return;
+        }
+        System.out.print(root.getData() + " ");
+        preOrderTraversalHelper(root.getLeft());
+        preOrderTraversalHelper(root.getRight());
+    }
+
+    public void inOrderTraversal(){
+        System.out.println("In order traversal");
+        inOrderTraversalHelper(this.root);
+        System.out.println("\n");
+    }
+    private void inOrderTraversalHelper(BSTNode root){
+        if (root == null){
+            return;
+        }
+        preOrderTraversalHelper(root.getLeft());
```

Comment:
I found function name is inOrder but recursive is calling to preOrder, is it right?